specutils/io/default_loaders/wcs_fits.py

```diff
@@ -157,12 +236,16 @@ def wcs1d_fits_writer(spectrum, file_name, hdu=0, update_header=False, **kwargs)
         raise ValueError(f'Only Spectrum1D objects with valid WCS can be written as wcs1d: {err}')
 
     # Verify spectral axis constructed from WCS
-    wl = spectrum.spectral_axis
-    dwl = (wcs.all_pix2world(np.arange(len(wl)), 0) - wl.value) / wl.value
-    if np.abs(dwl).max() > 1.e-10:
-        m = np.abs(dwl).argmax()
+    disp = spectrum.spectral_axis
+    # Not sure why the extra check is necessary for FITS WCS
```

Comment:
This comment does not inspire confidence. Can someone well-versed with spectrum WCS chime in here?

Reply:
The comment refers to my expectation for a WCS with a single, spectral dimension, that has a `spectral` (and thus also a `celestial`) attribute, to return the same world coordinates via that `wcs.spectral` as from `wcs` directly.
But that case is also handled separately in
https://github.com/astropy/specutils/blob/main/specutils/spectra/spectrum1d.py#L277-L284
although perhaps the coverage there is incomplete, don't know.